### PR TITLE
feat: add nightly build worflow

### LINKED ISSUE #21: Automatically verify (outgoing) links
Ruby library `htmlproofer` can be used to check/verify page links. Would be nice to include `htmlproofer` in the Travis (deploy) build.

### PR DESCRIPTION
**Pull request by mschilling**
This workflow:
* will automatically run every day at midnight
* includes a module to verify all links (internal + external)

Fixes #21 

Note: the nightly job is succeeding on node 14, but also failing on node 16. This failing job should NOT block this PR, instead it shows why a nightly build is useful.

### CODE REVIEW

File: .github/workflows/cicd.yml [hunk 1]
```diff
@@ -17,17 +17,17 @@ jobs:
           fetch-depth: '0'
           persist-credentials: false
 
-      - name: Use Node.js 12
+      - name: Use Node.js 14
```
Comment (cassshh) [inline .github/workflows/cicd.yml:20]: Don't we use a matrix here yet 🥲

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - cron: '0 2 * * *' # run at 2 AM UTC
+  workflow_dispatch:
+
+jobs:
+  nighlty:
+    name: Nightly build job
+
+    runs-on: ${{ matrix.os }}
+
+    continue-on-error: true
```
Comment (cassshh) [inline .github/workflows/nightly.yml:14]: Now it will also continue on error for node-14, which is what we don't want right?

**Review by cassshh — CHANGES_REQUESTED** [review]

File: .github/workflows/nightly.yml (hunk 2 above)
Comment (mschilling) [reply .github/workflows/nightly.yml:14]: Correct. 
- Long term we have to come up with a solution that fits our needs
- Short term: are you able to push a small change to improve on this? And if possible, can you also include the matrix change? (your previous comment)

File: .github/workflows/cicd.yml (hunk 1 above)
Comment (mschilling) [reply .github/workflows/cicd.yml:20]: No we didn't yet :sadpepe:

**Comment by mschilling** [comment]
@cassshh do we trust the implementation enough, so we can ignore the ci checks?

**Comment by cassshh** [comment]
Let's wait till tomorrow. Maybe we can also try using the GitHub CLI to run the action to get more info.

**Review by cassshh — APPROVED** [review]
🪄

**Review by cassshh — APPROVED** [review]

**Pull request merged**